src/main/java/net/coderbot/iris/gui/property/OptionProperty.java

```diff
@@ -0,0 +1,111 @@
+package net.coderbot.iris.gui.property;
+
+import com.google.common.collect.Lists;
+import net.coderbot.iris.Iris;
+import net.coderbot.iris.gui.GuiUtil;
+import net.coderbot.iris.gui.element.PropertyDocumentWidget;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.text.Text;
+
+import java.util.List;
+
+public abstract class OptionProperty<T> extends ValueProperty<T> {
+    protected T[] values;
+    protected int index;
+    protected final int defaultIndex;
+    protected final boolean isSlider;
+
+    public OptionProperty(T[] values, int defaultIndex, PropertyDocumentWidget document, String key, Text label, boolean isSlider) {
+        super(document, key, label);
+        this.values = values;
+        this.index = defaultIndex;
+        this.defaultIndex = defaultIndex;
+        this.isSlider = isSlider;
+    }
+
+    public void cycle() {
+        this.index++;
+        if(index >= values.length) index = 0;
+        this.valueText = null;
+    }
+
+    @Override
+    public T getValue() {
+        return values[index];
+    }
+
+    @Override
+    public void setValue(T value) {
+        List<T> vList = Lists.newArrayList(values);
+        if(vList.contains(value)) {
+            this.index = vList.indexOf(value);
+        } else {
+            Iris.logger.warn("Unable to set value of {} to {} - Invalid value!", key, value);
+            this.index = defaultIndex;
+        }
+        this.valueText = null;
+    }
+
+    protected boolean isButtonHovered(double mouseX, boolean entryHovered) {
+        return entryHovered && mouseX > cachedX + (cachedWidth * 0.6) - 7;
+    }
+
+    @Override
+    public boolean onClicked(double mouseX, double mouseY, int button) {
```

Comment:
I kinda feel this all is just a lost button here, I'd assume value property set this up earlier, but it seems more appropriate to whip up a `ValuePropertyButton` rather than make the property value also a button.

Reply:
OptionProperty is separate from ValueProperty. ValueProperty is a base class which is simply a property which stores a modifiable value. OptionProperty includes a UI for it all. I don't think a separate class for the button is necessary, however if anyone else has any input that would be appreciated.